cli/src/commands/redo.rs

```diff
@@ -28,6 +28,14 @@ use crate::ui::Ui;
 /// Redo the most recently undone operation
 ///
 /// This is the natural counterpart of `jj undo`.
+///
+/// Repeated invocations of `jj undo` and `jj redo` are meant to feel intuitive
+/// if you are used to the model of an "undo stack". However, the behavior is
+/// not completely identical in complicated cases. If interested in the details,
+/// experiment and compare the command output to `jj op log` output and/or see
+/// the comments in the source code.
+///
+/// See also `jj op restore` to explicitly restore an older operation by id.
```

Comment:
Maybe mention `op log` here as well, as in the documentation of `jj undo`? I think it's even more important with redo, because users might undo some important change, accidentally create a new operation and then fail to redo the important operation. Then they might ask themselves: "Is there some way I can redo the operation I just lost?" They have to find out about `op log` to even identify which operation to restore.

Reply:
It was mentioned right before, and I didn't want to repeat. Still, it's a good point that no matter how we rewrite the above paragraph, we want to make sure `op log` is mentioned (and perhaps more prominently than it currently is).